BatteriesTest/satisfying.lean

```diff
@@ -0,0 +1,23 @@
+import Batteries.Lean.SatisfiesM
+import Batteries.Data.Array.Monadic
+
+open Lean Meta Array Elab Term Tactic Command
+
+-- For now these live in the test file, as it's not really clear we want people relying on them!
```

Comment:
I think these instances should be next to the `MonadSatisfying` instances, since it makes no sense to have a `MonadSatisfying` instance without the `LawfulMonad` instance

Reply:
I made a separate file, but agree they can be lifted out of the tests.